mozilla/foundation.mozilla.org
# updated method to get topics for recent_blog_post filter

# Description



Link to sample test page:
Related PRs/issues: #8224 

The `RecentBlogEntries` block used to have a hard coded list of topics that were able to be used in it's "topic filter":

This PR introduces a minor change to the method that the topic filter uses in order to get its list of possible options, as well as the necessary migrations for the change. Now, the list is updated dynamically and includes any new topics that have been added to the DB.
 
# Checklist



**Tests**

**Changes in Models:**

**Documentation:**



# Steps to test

1. Pull this branch and run `inv new-db`
2. Log into the CMS, and visit a page that can use the `RecentBlogEntries` block. For a quick link, you can visit a campaign page at http://localhost:8000/cms/pages/87/edit/
3. Add a new `RecentBlogEntries` block to the body, and select the drop-down for the "Filter By Topic" section. Note that there are only a handful of options, which are the topics that are currently in the DB.
<img width="2086" alt="image" src="https://user-images.githubusercontent.com/18314510/195947513-2893edc7-76cf-4336-a4b5-5e9fc8e4ffee.png">

4. Visit http://localhost:8000/cms/snippets/wagtailpages/blogpagetopic/add/?locale=en, and create a new Topic.

5. Go back to the `RecentBlockEntries` at http://localhost:8000/cms/pages/87/edit/, and take a look at the Topic options again. Note that the new topic is in the list.

<img width="2086" alt="Screen Shot 2022-10-14 at 2 36 52 PM" src="https://user-images.githubusercontent.com/18314510/195947902-4e8cea49-1aa3-4324-8a3f-63cf7e22185e.png">


6. If everything is working as expected, testing is complete!
